### PR TITLE
misc: replace 'whitelist' variants with 'allowlist'.

### PR DESCRIPTION
**Pull request by mkrawczuk**
For the record: this PR was inspired by https://github.com/nodejs/node/pull/33813

### CODE REVIEW

File: packages/remark-lint-maximum-line-length/index.js [hunk 1]
```diff
@@ -157,18 +157,18 @@ function maximumLineLength(tree, file, option) {
...
     if (!generated(node)) {
-      whitelist(start(node).line - 1, end(node).line)
+      allowlist(start(node).line - 1, end(node).line)
     }
   }
 
-  // Whitelist from `initial` to `final`, zero-based.
-  function whitelist(initial, final) {
+  // Allowlist from `initial` to `final`, zero-based.
+  function allowlist(initial, final) {
```
Comment (ChristianMurphy) [inline packages/remark-lint-maximum-line-length/index.js:171]:
```suggestion
  function allowList(initial, final) {
```

File: packages/remark-lint-maximum-line-length/index.js [hunk 2]
```diff
@@ -157,18 +157,18 @@ function maximumLineLength(tree, file, option) {
...
 
-    whitelist(initial.line - 1, final.line)
+    allowlist(initial.line - 1, final.line)
   }
 
   function ignore(node) {
     /* istanbul ignore else - Hard to test, as we only run this case on `position: true` */
     if (!generated(node)) {
-      whitelist(start(node).line - 1, end(node).line)
+      allowlist(start(node).line - 1, end(node).line)
```
Comment (ChristianMurphy) [inline packages/remark-lint-maximum-line-length/index.js:166]:
```suggestion
      allowList(start(node).line - 1, end(node).line)
```

File: packages/remark-lint-maximum-line-length/index.js [hunk 3]
```diff
@@ -157,18 +157,18 @@ function maximumLineLength(tree, file, option) {
       return
     }
 
-    whitelist(initial.line - 1, final.line)
+    allowlist(initial.line - 1, final.line)
```
Comment (ChristianMurphy) [inline packages/remark-lint-maximum-line-length/index.js:160]:
```suggestion
    allowList(initial.line - 1, final.line)
```

**Comment by wooorm** [comment]
Thanks @Mateusz, sorry this was in there!

**Comment by mkrawczuk** [comment]
Nothing to apologize for. I always felt that allow/deny is just self-explanatory and am glad more people start to agree with it.

**Pull request merged**